proposed/0000-safecoin-market-price.md

```diff
@@ -0,0 +1,74 @@
+- Feature Name: SafeCoin_Market_Price
+- Type: New feature
+- Related components: maidsafe_vault, maidsafe_client, maidsafe_types
+- Start Date: 19-07-2015
+- RFC PR: (leave this empty)
+- Issue number: (leave this empty)
+
+# Summary
+
+This RFC is based on the concerns I’ve read in the MaidSafe forum for the last six months. These are some of them:
+
+* https://forum.safenetwork.io/t/lowest-put-price-discovery-and-new-safecoin-issuance/3819
+* https://forum.safenetwork.io/t/cost-of-storing-data-on-safe/4285
+* https://forum.safenetwork.io/t/the-price-of-safecoin-the-economics-behind-it/844
+
+This RFC is only another point of view (maybe not doable, feel free to reject this RFC) that breaks a lot the main design. Maybe this RFC can be decomposed in several RFC’s, smaller and feasible.
+
+
+# Motivation
+
+My first concern is how much cost to store a chunk of data (cost in terms of USD,EUR or BTC). No one is able to answer this question accurately. I’m thinking in a true Offer/Demand market, where you establish how much you want to pay to store data, and in the other side, the vaults establish how much they want to get paid to store that data.
+
```

Comment:
Ah this is where the network calculates in real time that cost. Otherwise it's not going to be able to give the most efficient price. It requires a lot of explanation, here is a massively simplified view of what happens and what's being tested. 

Data is stored in 3 locations (ignore replication etc.) Original Backup and Sacrificial
Sacrificial Data is the one that gets removed to store other types 

Sacrificial is also the network storing 50% more copies than it needs (very healthy) 

As each sacrifical is stored the Farming Rate doubles  (twice as hard to get a safecoin)
As each Sacrifical cannot be stored or is removed farming rate halves (farming more profitable)

Farming rate is then used to calculate cost of storage So a user will pay something close to the 5th root of farming rate per chunk (this is to be tested)

So this means the network calculates cost on network health and in `real time` which is very much designed to provide resources at the minimum price people (farmers) will provide those resources. Another push is farmer efficiency so people provide unused resources (computer already on share unused b/w/ hdd and cpu) which should drive costs of resource purchase to an absolute minimum. 

External safecoin pricing is separate but linked. If farmers get $X per hour they may wish to farm, if it goes to 1/X they may choose not to, the network will rebalance automatically regardless of external cost of safecoin. All it discovers is the farming rate required to get more resources, it does no know why that is the rate but calculates based on the rate required to attract/deter farmers whatever that may be.